app/waf/src/controllers/wafpolicy.controller.ts

```diff
@@ -120,6 +129,94 @@ export class WafpolicyController extends BaseController {
     );
   }
 
+  @post(prefix + '/wafpolicies/{id}/adcs/{adcId}', {
+    responses: {
+      '204': Schema.emptyResponse('Uploading WAF Policy resource'),
+      '404': Schema.notFound('Can not find trustedDeivceId'),
+      '422': Schema.unprocessableEntity(' Adc is not trusted'),
+    },
+  })
+  async wafpolicyUpload(
+    @param(Schema.pathParameter('id', 'WAF Policy resource ID')) id: string,
+    @param(Schema.pathParameter('adcId', 'ADC resource ID')) adcId: string,
+  ): Promise<void> {
+    let wafpolicy: Wafpolicy = await this.wafpolicyRepository.findById(id, {
+      where: {
+        and: [
+          {or: [{tenantId: await this.tenantId}, {public: true}]},
+          {id: id},
+        ],
+      },
+    });
+
+    let adc: Adc = await this.adcRepository.findById(adcId, undefined, {
```

Comment:
`findById` and `find` are different. `findById` will throw error while `find` will return []. 

So in this controller function`wafpolicyUpload`, there should be `try .. catch` to handle possible exceptions.